### DISCUSSION TITLE
WebContainers

### DISCUSSION THREAD

**Discussion by aehlke** [live-codes/livecodes#575] [Ideas]
I missed this announcement that StackBlitz made WebContainers open for custom reuse outside their platform: https://blog.stackblitz.com/posts/webcontainer-api-is-here/

This would let you run node.js based tools, or anything requiring a VM - could really expand what you offer. I haven't dug into what exactly they made public here to understand if it's sufficient for you though.

**Comment by hatemhosny** [comment]
Thank you @aehlke 

StackBlitz's WebContainers are indeed quite powerful, with different advantages and disadvantages. 
My major concern is that it is not open-source. I prefer not using that in the core of LiveCodes. 
However, that can be a good use for [plugins](https://github.com/live-codes/livecodes/issues/481) when that is supported (which I'm serious about adding).
